Drivers/MotorChannel/Inc/ZP_D_PWMChannel.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef ZP_D_PWM_CHANNEL_HPP_
+#define ZP_D_PWM_CHANNEL_HPP_
+
+#include "main.h"
+
+#include "ZP_D_MotorChannel.hpp"
+
+class PWMChannel : public MotorChannel {
+    public:
+        PWMChannel(uint16_t pin_num, GPIO_TypeDef* port, TIM_HandleTypeDef* timer,
+					uint16_t timer_channel,uint32_t clock_frequency);
+
+        void setup(void);
+        void set(uint8_t percent);
+    private:
+        uint16_t pin_num_;
+        GPIO_TypeDef* port_;
+        TIM_HandleTypeDef *timer_;     //handle to the timer
+        uint16_t timer_channel_;       //channel of the timer
+        uint32_t clock_frequency_;     //system clock frequency
```

Comment:
This is likely something we can determine at compile time and don't really need to initialize in the constructor is it?
A bunch of these can also be `const`